### DISCUSSION TITLE
Getting GET_PED_NEARBY_PEDS to work / Suggestion for YimMenu

### DISCUSSION THREAD

**Discussion by MindOfBog** [YimMenu/YimMenu#3645] [Ideas]
`PED.GET_PED_NEARBY_PEDS(ped, struct, ignore)` 
This is a native function that I'm unable to get working under LUA, normally this required a struct which contains the amount of peds (I think) and also an array which will be filled with nearby peds. This isn't really possible under lua. I have tried a few different ways but none worked, I'm guessing it isn't implemented in a usable way yet?

I have done some digging and found a neat implementation for it in a Lua plugin for ScriptHookV where it works like this:

`local array, size = PED.GET_PED_NEARBY_PEDS(ped, count, ignore)`

basically it's the same, except you provide the count (instead of a struct), which will then be used by yim to turn it into the struct the native function needs. 

then the function returns both: 
the array (which would normally be provided and filled by the function)
the size (which is the normal return value from this function)

**Comment by MindOfBog** [comment]
This same implementation could also be used for `GET_PED_NEARBY_VEHICLES`

**Comment by xesdoog** [comment]
Read the [Lua docs](https://github.com/YimMenu/YimMenu/blob/master/docs/lua/tables/entities.md).

`all_peds = entities.get_all_peds_as_handles()` -- returns a table with all ped handles
`all_vehicles = entities.get_all_vehicles_as_handles()` -- returns a table with all vehicle handles
`all_objects = entities.get_all_objects_as_handles()` -- returns a table with all object handles

To get nearby peds, iterate over the `all_peds` table and [compare the coords of each value to yours](https://nativedb.dotindustries.dev/gta5/natives/0x2A488C176D52CCA5?search=vdist).

**Reply by MindOfBog** [reply]
Those are quite convenient and probably more useful for what I need than the nearby peds function (again sorry for not having read up on the docs much lol)

I think my suggestion would still be cool to have, since those functions are not really useful for anything in their current state (talking about `PED.GET_PED_NEARBY_PEDS`)

Thanks again

**Comment by xiaoxiao921** [comment]
These functions are usable, you can allocate the array struct with the lua api with memory.allocate , the struct is defined such as the pastebin link from the fivem native doc

**Reply by MindOfBog** [reply]
I did actually see that in the fivem docs but I had no idea you can create structs in lua like this, interesting
